tools/gulp/tasks/screenshots.ts

```diff
@@ -0,0 +1,190 @@
+}
+
+function updateTravis(database: admin.database.Database,
+                      prNumber: string): admin.Promise<void> {
+  return database.ref(FIREBASE_REPORT).child(prNumber).update({
+    'commit': process.env['TRAVIS_COMMIT'],
+    'sha': process.env['TRAVIS_PULL_REQUEST_SHA'],
+    'travis': process.env['TRAVIS_JOB_ID'],
+  });
+}
+
+/** Get a list of filenames from firebase database. */
+function getScreenFilenames(database: admin.database.Database): admin.Promise<string[]> {
+  return database.ref(FIREBASE_FILELIST).once('value')
+      .then((snapshots: admin.database.DataSnapshot) => {
+    return snapshots.val();
+  });
+}
+
+/** Upload a list of filenames to firebase database as gold. */
+function setScreenFilenames(database: admin.database.Database,
+                            reportKey?: string): admin.Promise<void> {
+  let filenames: string[] = [];
+  readdirSync(SCREENSHOT_DIR).map((file: string) => {
+    let fullName = path.join(SCREENSHOT_DIR, file);
+    let key = file.replace('.screenshot.png', '');
+    if (!statSync(fullName).isDirectory() && key) {
+      filenames.push(file);
+    }
+  });
+  let filelistDatabase = reportKey ?
+    database.ref(FIREBASE_REPORT).child(reportKey).child('filenames') :
+    database.ref(FIREBASE_FILELIST);
+  return filelistDatabase.set(filenames);
+}
+
+/**
+ * Upload screenshots to google cloud storage.
+ * @param {string} reportKey - The key used in firebase. Here it is the PR number.
+ *   If there's no reportKey, we will upload images to 'golds/' folder
+ * @param {string} mode - Can be 'test' or 'diff' or null.
+ *   If the images are the test results, mode should be 'test'.
+ *   If the images are the diff images generated, mode should be 'diff'.
+ *   For golds mode should be null.
+ */
+function uploadScreenshots(reportKey?: string, mode?: 'test' | 'diff') {
+  let bucket = openScreenshotsCloudStorage();
+
+  let promises: admin.Promise<void>[] = [];
+  let localDir = mode == 'diff' ? `${SCREENSHOT_DIR}/diff` : SCREENSHOT_DIR;
+  readdirSync(localDir).map((file: string) => {
+    let fileName = path.join(localDir, file);
+    let key = file.replace('.screenshot.png', '');
+    let destination = (mode == null || !reportKey) ?
+      `golds/${file}` : `screenshots/${reportKey}/${mode}/${file}`;
+
+    if (!statSync(fileName).isDirectory() && key) {
+      promises.push(bucket.upload(fileName, { destination: destination }));
+    }
+  });
+  return admin.Promise.all(promises);
+}
+
+/** Check whether the directory exists. If not then create one. */
+function _makeDir(dirName: string) {
+  if (!existsSync(dirName)) {
+    mkdirSync(dirName, '744');
+  }
+}
+
+/** Download golds screenshots. */
+function downloadAllGolds(
+    filenames: string[], database: admin.database.Database,
+    reportKey: string): admin.Promise<boolean> {
+  _makeDir(`${SCREENSHOT_DIR}/golds`);
+
+  return admin.Promise.all(filenames.map((filename: string) => {
```

Comment:
Do we actually need to store the list of gold filenames? I think `google-cloud-node` allows you to query a list of files (See [here](https://googlecloudplatform.github.io/google-cloud-node/#/docs/storage/0.6.0/storage/bucket?method=getFiles))
```ts
let bucket = openScreenshotsBucket();

bucket.getFiles({ prefix: 'golds/' }).then(function(data) {
  var files = data[0];
});


```